src/main/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObject.scala

```diff
@@ -0,0 +1,150 @@
+import io.smartdatalake.definitions.DateColumnType.DateColumnType
+import io.smartdatalake.util.hdfs.{HdfsUtil, PartitionValues}
+import io.smartdatalake.util.hive.HiveUtil
+import io.smartdatalake.util.misc.{AclDef, AclUtil}
+import io.smartdatalake.workflow.connection.HiveTableConnection
+import org.apache.hadoop.fs.FileSystem
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Row, SaveMode, SparkSession}
+
+import scala.collection.JavaConverters._
+
+case class DeltaLakeTableDataObject(override val id: DataObjectId,
+                                    path: String,
+                                    override val partitions: Seq[String] = Seq(),
+                                    analyzeTableAfterWrite: Boolean = false,
+                                    dateColumnType: DateColumnType = DateColumnType.Date,
+                                    override val schemaMin: Option[StructType] = None,
+                                    override var table: Table,
+                                    numInitialHdfsPartitions: Int = 16,
+                                    saveMode: SaveMode = SaveMode.Overwrite,
+                                    acl: Option[AclDef] = None,
+                                    connectionId: Option[ConnectionId] = None,
+                                    override val metadata: Option[DataObjectMetadata] = None)
+                                   (@transient implicit val instanceRegistry: InstanceRegistry)
+  extends TransactionalSparkTableDataObject with CanHandlePartitions {
+
+  /**
+   * Connection defines db, path prefix (scheme, authority, base path) and acl's in central location
+   */
+  private val connection = connectionId.map(c => getConnection[HiveTableConnection](c))
+
+  // prepare final path and table
+  @transient private[workflow] lazy val hadoopPath = HdfsUtil.prefixHadoopPath(path, connection.map(_.pathPrefix))
+  @transient private var filesystemHolder: FileSystem = _
+  def filesystem(implicit session: SparkSession): FileSystem = {
+    if (filesystemHolder == null) {
+      filesystemHolder = HdfsUtil.getHadoopFsFromSpark(hadoopPath)
+    }
+    filesystemHolder
+  }
+  table = table.overrideDb(connection.map(_.db))
+  if (table.db.isEmpty) {
+    throw ConfigurationException(s"($id) db is not defined in table and connection for dataObject.")
+  }
+
+  override def prepare(implicit session: SparkSession): Unit = {
+    require(isDbExisting, s"($id) DB ${table.db.get} doesn't exist (needs to be created manually).")
+  }
+
+  override def getDataFrame(partitionValues: Seq[PartitionValues] = Seq())(implicit session: SparkSession): DataFrame = {
+    val df = session.read.format("delta").load(hadoopPath.toString)
+    validateSchemaMin(df)
+    df
+  }
+
+  override def init(df: DataFrame, partitionValues: Seq[PartitionValues])(implicit session: SparkSession): Unit = {
+    // create tables if possible
+    prepare(session)
+    if (!isTableExisting) {
```

Comment:
Delta Lake isn't connected to Hive Metastore for now... it's file base Spark API. We cannot check isTableExisting.

Reply:
Done: https://github.com/smart-data-lake/smart-data-lake/pull/13/commits/7c147455d1d452fd9b69d429fbd18e54b73c4da7